### PR TITLE
refactor(frontend): Only functional components

### PR DESCRIPTION
**Pull request by luixo**
I believe a modern React app should move to use only functional components as much as possible, as using hooks making code significantly more readable.

### CODE REVIEW

**Review by luixo — COMMENTED** [review]
Hint: to make review more doable press `hide whitespace` under gearbox icon in diff section

File: frontend/src/components/nodes/__test__/__snapshots__/NodesEpoch.test.tsx.snap [hunk 1]
```diff
@@ -67,16 +67,7 @@ exports[`<NodesEpoch /> renders 1`] = `
         >
           <svg
             className="rc-progress-circle progress-bar-content"
-            label={
```
Comment (luixo) [inline frontend/src/components/nodes/__test__/__snapshots__/NodesEpoch.test.tsx.snap:70]: TODO: find out what happened here

File: frontend/src/components/utils/LocalizeWrapper.tsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { PureComponent } from "react";
 import { setI18N } from "../../libraries/language";
 import { LocalizeContextProps, withLocalize } from "react-localize-redux";
 
+// TODO: migrate from react-localize-redux to a package that supports hooks
```
Comment (luixo) [inline frontend/src/components/utils/LocalizeWrapper.tsx:5]: @ reviewers: please note

File: frontend/src/components/utils/Timer.tsx [hunk 3]
```diff
@@ -1,57 +1,23 @@
-import { PureComponent } from "react";
```
Comment (luixo) [inline frontend/src/components/utils/Timer.tsx]: This is very model of how using hooks may improve dev's life

File: frontend/src/hooks/use-near-network.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { useContext } from "react";
+import { NetworkContext } from "src/context/NetworkProvider";
+import { NearNetwork } from "src/libraries/config";
+
+export const useNearNetwork = (): {
```
Comment (luixo) [inline frontend/src/hooks/use-near-network.ts]: @frol The hook I was talking about

File: frontend/src/components/nodes/__test__/__snapshots__/NodesEpoch.test.tsx.snap (hunk 1 above)
Comment (luixo) [reply frontend/src/components/nodes/__test__/__snapshots__/NodesEpoch.test.tsx.snap:70]: Yep, we probably don't need to pass attributes "type" and "label" to svg node :)

File: frontend/src/pages/accounts/[id].tsx [hunk 5]
```diff
@@ -15,9 +14,10 @@ import Content from "../../components/utils/Content";
 import TransactionIcon from "../../../public/static/images/icon-t-transactions.svg";
 
 import { Translate } from "react-localize-redux";
-import { NextPageContext } from "next";
+import { NextPage } from "next";
+import { useNearNetwork } from "src/hooks/use-near-network";
```
Comment (frol) [inline frontend/src/pages/accounts/[id].tsx]: 1. These do not follow the contribution guide. Let's move the imports into the appropriate groups.
2. Should we change all relative imports that reach `src` folder to `src/` imports? (we should mention that in the contributing guide)

**Review by frol — APPROVED** [review]

**Review by frol — CHANGES_REQUESTED** [review]
Something wrong has happened on https://near-explorer-experimental-frontend-with-3z4t.onrender.com/nodes/validators

image

File: frontend/src/components/transactions/TransactionAction.tsx [hunk 6]
```diff
@@ -13,57 +13,37 @@ export interface Props {
...
-
-  componentDidMount() {
-    this.fetchStatus();
+const TransactionAction: FC<Props> = ({ transaction, viewMode = "sparse" }) => {
+  const [status, setStatus] = useState();
+  useEffect(() => {
+    new TransactionsApi()
+      .getTransactionStatus(transaction.hash, transaction.signerId)
+      .then(setStatus);
+  }, []);
```
[the remark this answers is not in the document]
Comment (luixo) [reply frontend/src/components/transactions/TransactionAction.tsx:22]: That's a good catch, I missed it.
We need `rules-of-hooks` lint rule for that.

File: frontend/src/pages/accounts/[id].tsx (hunk 5 above)
Comment (luixo) [reply frontend/src/pages/accounts/[id].tsx]: I believe there is only one way that has minimal frustration to enforce any rules on import order / styles - linter (with fix).
Should I make it as in contribution guide in this PR or should we wait until I'll incorporate linter with import-order rule?

**Comment by shelegdmitriy** [comment]
@luixo I found this error after navigating to Accounts page
```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of `FlipMove`.
AccountRow@webpack-internal:///./src/components/accounts/AccountRow.tsx:31:21
div
FlipMove@webpack-internal:///./node_modules/react-flip-move/dist/react-flip-move.es.js:720:19
FlipMovePropConverter@webpack-internal:///./node_modules/react-flip-move/dist/react-flip-move.es.js:282:21
FlipMoveEx@webpack-internal:///./src/components/utils/FlipMove.tsx:69:21
Accounts@webpack-internal:///./src/components/accounts/Accounts.tsx:34:17
div
div
InfiniteScroll@webpack-internal:///./node_modules/react-infinite-scroll-component/dist/index.es.js:199:28
WrappedTranslate@webpack-internal:///./node_modules/react-localize-redux/es/Translate.js:36:20
Translate
Wrapper/<@webpack-internal:///./src/components/utils/ListHandler.tsx:29:66
AccountsWrapper
div
Container<@webpack-internal:///./node_modules/react-bootstrap/esm/Container.js:19:18
div
Container<@webpack-internal:///./node_modules/react-bootstrap/esm/Container.js:19:18
Content@webpack-internal:///./src/components/utils/Content.tsx:13:17
AccountsPage@webpack-internal:///./src/pages/accounts/index.tsx:24:53
LocalizeWrapper@webpack-internal:///./src/components/utils/LocalizeWrapper.tsx:104:24
LocalizedComponent@webpack-internal:///./node_modules/react-localize-redux/es/withLocalize.js:29:22
DatabaseProvider@webpack-internal:///./src/context/DatabaseProvider.tsx:24:62
div
NetworkProvider@webpack-internal:///./src/context/NetworkProvider.tsx:18:13
LocalizeProvider@webpack-internal:///./node_modules/react-localize-redux/es/LocalizeProvider.js:27:20
App@webpack-internal:///./src/pages/_app.tsx:107:21
StyleRegistry@webpack-internal:///./node_modules/styled-jsx/dist/stylesheet-registry.js:231:28
ErrorBoundary@webpack-internal:///./node_modules/@next/react-dev-overlay/lib/internal/ErrorBoundary.js:26:47
ReactDevOverlay@webpack-internal:///./node_modules/@next/react-dev-overlay/lib/internal/ReactDevOverlay.js:90:20
Container@webpack-internal:///./node_modules/next/dist/client/index.js:331:24
AppContainer@webpack-internal:///./node_modules/next/dist/client/index.js:770:20
Root@webpack-internal:///./node_modules/next/dist/client/index.js:891:21
```

File: frontend/src/components/accounts/Accounts.tsx [hunk 7]
```diff
@@ -1,56 +1,42 @@
...
+
+const AccountsList = ListHandler({
+  Component: Accounts,
+  category: "Account",
+  paginationIndexer: (items) => ({
+    endTimestamp: items[items.length - 1].createdAtBlockTimestamp
+      ? items[items.length - 1].createdAtBlockTimestamp || undefined
+      : undefined,
+    accountIndex: items[items.length - 1].accountIndex,
+  }),
```
Comment (shelegdmitriy) [inline frontend/src/components/accounts/Accounts.tsx:41]: I like it! So now it's much intuitive of what we are doing with data

Comment (luixo) [reply frontend/src/components/accounts/Accounts.tsx:41]: What exactly? Refactoring indexer out of `List` itself to `List` implemetations?

**Comment by luixo** [comment]
> Something wrong has happened on https://near-explorer-experimental-frontend-with-3z4t.onrender.com/nodes/validators

The wrong is called `any` in props :) Thanks for catching that!

**Comment by luixo** [comment]
> I found this error after navigating to Accounts page

Fixed it.

**Review by frol — APPROVED** [review]

**Pull request merged**